src/plugin/file/kml/kmlexport.js

```diff
@@ -57,8 +57,14 @@ plugin.file.kml.export.getBalloonOptions = function(feature) {
 plugin.file.kml.export.getRotationColumn = function(feature) {
   if (feature) {
     var layerConfig = os.style.getLayerConfig(feature);
-    if (layerConfig && layerConfig[os.style.StyleField.SHOW_ROTATION]) {
-      return layerConfig[os.style.StyleField.ROTATION_COLUMN];
+    if (layerConfig) {
+      if (layerConfig[os.style.StyleField.REPLACE_STYLE]) {
+        if (layerConfig[os.style.StyleField.SHOW_ROTATION]) {
+          return layerConfig[os.style.StyleField.ROTATION_COLUMN];
+        }
+      } else if (feature.get(os.style.StyleField.SHOW_ROTATION)) {
+        return /** @type {string} */ (feature.get(os.style.StyleField.ROTATION_COLUMN));
+      }
```

Comment:
This only uses the layer config when `REPLACE_STYLE` is true. That is not correct. I think you want something more like this:
```
var rotColumn = layerConfig[...];
if (!layerConfig[REPLACE_STYLE]) {
  rotColumn = rotColumn || feature.get(...);
}
```

Reply:
Doesn't that ignore if the show rotation box is checked or not though?
The kml exports with rotation whether or not that box is checked now ... I guess that could be tracked separately (ie remember the rotation column selection even if it isn't used)

Reply:
> Doesn't that ignore if the show rotation box is checked or not though?

It is just pseudocode. You'll need to add that in too.

Reply:
This function is only concerned with getting the column, and doesn't need to consider the `SHOW_ROTATION` value. Code using this function can choose whether or not to do something with that column based on `SHOW_ROTATION`.